File: api/src/main/java/fr/epf/speedycart/api/model/Shop.java
package fr.epf.speedycart.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PastOrPresent;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import org.hibernate.validator.constraints.Length;

import java.time.LocalDateTime;

@Entity
@Data
public class Shop {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Pattern(regexp = "[A-Za-z ]{1,50}",
            message = "Name must be 1-50 characters and only A-Z or a-z")
    @NotNull(message = "Name can not be null")
    private String name;

    @Length(min = 1, max = 300,
            message = "Description must be 1-300 characters")
    private String description;

    @PastOrPresent(message = "Active Since must be a past or the current date")
    private LocalDateTime activeSince;

    @FutureOrPresent(message = "Disable Since must be a future or the current date")
    private LocalDateTime disableSince;

    @Pattern(regexp = "[0-9]{14}",
            message = "Siret must be 14 characters and 0-9")
    @NotNull(message = "Siret can not be null")
    private String siret;

    @OneToOne
    @JoinColumn(name = "address_id", unique = true)
    @NotNull(message = "Adress can not be null")
    private Address address;
}

File: app/app/src/main/java/fr/epf/min1/speedycart/data/Shop.kt
package fr.epf.min1.speedycart.data

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.util.Date

@Parcelize
data class Shop(
    val shopId: Long,
    val name: String,
    val description: String,
    val activeSince: Date?,
    val disableSince: Date?,
    val siret: String,
    val address: Address
    ) : Parcelable{

    companion object{
        fun generate1Shop() = Shop(12,
            "Intermarché",
            "Ceci est une description",
            Date(1644677500000),
            null,
            "818 582 496 00027",
            Address.generate1Address()
        )

        fun generateListShop(): List<Shop>{
            val shopList = ArrayList<Shop>()
            shopList.add(Shop(
                1,
                "Intermarché",
                "Ceci est une description",
                Date(1644677500000),
                null,
                "818 582 496 00027",
                Address.generate1Address()
            ))
            shopList.add(Shop(
                2,
                "Intermarché2",
                "Ceci est une description",
                Date(1644677500000),
                null,
                "818 582 496 00027",
                Address.generate1Address()
            ))
            shopList.add(Shop(
                3,
                "Intermarché3",
                "Ceci est une description",
                Date(1644677500000),
                null,
                "818 582 496 00027",
                Address.generate1Address()
            ))
            shopList.add(Shop(
                4,
                "Intermarché4",
                "Ceci est une description",
                Date(1644677500000),
                null,
                "818 582 496 00027",
                Address.generate1Address()
            ))
            return shopList.toList()

        }
    }
}